### ISSUE TITLE
afterRemote hook not called on `download`

### ISSUE THREAD

**Issue by dddicillo** [strongloop/loopback-component-storage#59]
I receive no console output with the following lines:
```javascript
Container.afterRemote('download',
    function(ctx, affectedModelInstance, next) {
        console.log('Downloaded');
        next();
    }
);
```
I'm trying to encrypt files before sending and decrypt afterwards.

**Comment by alexthewilde** [comment]
+1

**Comment by diegobent** [comment]
+1
@dddicillo I'm trying to do the same operations. Have you found a solution?

**Comment by dddicillo** [comment]
@diegobent Unfortunately I wasn't able to resolve this issue. I messed around creating a custom endpoint that was functionally identical to the download endpoint, but it proved to be more work than I was willing to do at the time.

**Comment by richardpringle** [comment]
@dddicillo @alexthewilde @diegobent 

Sorry about the delayed response to this issue, this repo hasn't received enough love (support) lately.

The container model actually doesn't quite function the same way as a LoopBack PersistedModel for instance. I think the best thing to do to solve your issue would be to create a middleware function that is registered to the `"download"` endpoint. 

I will throw a feature request label on the issue for now, although it might take a while to implement as there not a lot of work being done on this module in particular. In general, we are working on some additional customization features that should make things like this easier to implement. 

Good luck, again sorry about the neglect, and happy coding!

**Comment by richardpringle** [comment]
UPDATE

Sorry, I hadn't played around with this enough. Definitely a bug. Since it's a bug, the above recommendation most likely will not work. 

Throwing the bug label on this one, hopefully will get fixed soon.

**Comment by diegobent** [comment]
thanks guys! at the moment I've created a custom endpoint and I use another module.
@richardpringle thank you for your work!

**Comment by richardpringle** [comment]
@dddicillo @alexthewilde @diegobent 

Fix has been landed, see #143. It was quite simple...sorry for taking so long to get to the issue.

Also see @bajtos' comment [here](https://github.com/strongloop/loopback-component-storage/pull/143#issuecomment-240329703) in case you want to use `download` without loopback-component-storage.

Cheers
